Modules/Core/Common/include/itkAnatomicalOrientation.h

```diff
@@ -0,0 +1,543 @@
+    SAL = m_OrientationValue<CoordinateEnum::SuperiorToInferior,
+                             CoordinateEnum::AnteriorToPosterior,
+                             CoordinateEnum::LeftToRight>,
+
+    PIR = m_OrientationValue<CoordinateEnum::PosteriorToAnterior,
+                             CoordinateEnum::InferiorToSuperior,
+                             CoordinateEnum::RightToLeft>,
+    PSR = m_OrientationValue<CoordinateEnum::PosteriorToAnterior,
+                             CoordinateEnum::SuperiorToInferior,
+                             CoordinateEnum::RightToLeft>,
+    AIR = m_OrientationValue<CoordinateEnum::AnteriorToPosterior,
+                             CoordinateEnum::InferiorToSuperior,
+                             CoordinateEnum::RightToLeft>,
+    ASR = m_OrientationValue<CoordinateEnum::AnteriorToPosterior,
+                             CoordinateEnum::SuperiorToInferior,
+                             CoordinateEnum::RightToLeft>,
+    PIL = m_OrientationValue<CoordinateEnum::PosteriorToAnterior,
+                             CoordinateEnum::InferiorToSuperior,
+                             CoordinateEnum::LeftToRight>,
+    PSL = m_OrientationValue<CoordinateEnum::PosteriorToAnterior,
+                             CoordinateEnum::SuperiorToInferior,
+                             CoordinateEnum::LeftToRight>,
+    AIL = m_OrientationValue<CoordinateEnum::AnteriorToPosterior,
+                             CoordinateEnum::InferiorToSuperior,
+                             CoordinateEnum::LeftToRight>,
+    ASL = m_OrientationValue<CoordinateEnum::AnteriorToPosterior,
+                             CoordinateEnum::SuperiorToInferior,
+                             CoordinateEnum::LeftToRight>
+
+  };
+
+
+  /** \brief Initialize with CoordinateEnum's from separate axes.
+   *
+   * If multiple CoordinateEnums are from the same axes then the Orientation value is INVALID.
+   */
+  AnatomicalOrientation(CoordinateEnum primary, CoordinateEnum secondary, CoordinateEnum tertiary);
+
+  AnatomicalOrientation(ToEnum orientation)
+    : m_Value(orientation)
+  {}
+
+
+#ifndef ITK_FUTURE_LEGACY_REMOVE
+  /** \brief Conversion from Legacy SpatialOrientation
+   *
+   * @param legacyOrientation
+   */
+#  if defined(ITK_LEGACY_REMOVE) && !defined(ITK_LEGACY_SILENT)
+  [[deprecated("Use the AnatomicalOrientation::FromEnum type instead.")]]
+#  endif
+  AnatomicalOrientation(LegacyOrientationType legacyOrientation);
+#endif
+
+  AnatomicalOrientation(FromEnum orientation);
+
+  explicit AnatomicalOrientation(const DirectionType & d)
+    : m_Value(DirectionCosinesToOrientation(d))
+  {}
+
+  operator ToEnum() const { return m_Value; }
+
+  std::string
+  GetAsToStringEncoding() const;
+
+  std::string
+  GetAsFromStringEncoding() const;
+
+  static AnatomicalOrientation
+  CreateFromToStringEncoding(std::string str);
+
+  static AnatomicalOrientation
+  CreateFromFromStringEncoding(std::string str);
```

Comment:
"FromFrom", is that intended? It looks a bit funny 🤷

Reply:
Yes. There is "ToEnum" and "FromEnum" types defined, the the string representation are called "To String Encoding" and "From String Encoding". Then "Create From X" was the prefix for these explicit constructor-like methods.

Do you have another suggestion?

Reply:
> Do you have another suggestion?

Sorry, no! I haven't entirely studied your proposal, I just wanted to be sure that "FromFrom" is not a typo! (Other users might also find it confusing, I don't know.)

By the way I find it hard to review a PR of 1000+ changed lines of code and 10+ commits. (Especially when it comes to _approving_ the entire PR.) I don't know about this one specifically, but in general I would suggest to try to keep PRs a bit smaller. (And then do follow-up PRs, if necessary.) Anyway, that's just a general comment, it may be too late for this specific PR!

Reply:
Yes, this PR has gotten a big unwieldy and likely should be squashed into just a couple commits. But due to the requests, I wanted to keep the change set until it was close, which I guess it is now.

I'd recommended just reviewing the new itkAnatomicalOrientation.h file closely.

Reply:
Reading this class and the implementation for the first time, everything is quite good, but I was also confused by the `GetAsToStringEncoding`, `GetAsFromStringEncoding`, `CreateFromToStringEncoding`, `CreateFromFromStringEncoding`. Possible improvement: `GetPositiveStringEncoding`, `GetNegativeStringEncoding`, CreateFromPositiveStringEncoding`, `CreateFromNegativeStringEncoding`.

Reply:
That seems like an improved convention, it would also follow to rename "ToEnum"->"PositiveEnum" and "FromEnum"->"NegativeEnum".